### PR TITLE
Bug 1259763 - Reader mode omits opening paragraph on CNN articles, r=Gijs

### PR DESCRIPTION
**Pull request by evanxd**
Added test and found out somehow the `<h2>The U.S. has long been heralded as a land of opportunity -- a place where anyone can succeed regardless of the economic class they were born into.</h2>` node in the `source.html` is just removed by some kind of reason. Continue to investigate the root cause...

### CODE REVIEW

**Comment by gijsk** [comment]
See https://github.com/mozilla/readability/issues/281#issuecomment-208893497 .

I expect we should avoid removing the `<h2>` if it is substantially different from the title of the page that we detect.

**Comment by evanxd** [comment]
@gijsk 
I updated patch for the comments. Could you take a look again? Thanks.

File: test/test-pages/medium-1/expected.html [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <div id="readability-page-1" class="page">
     <div class="section-inner layoutSingleColumn">
+        <h2 name="3c62" id="3c62" data-align="center" class="graf--h2">Open Journalism Project:</h2>
         <h4 name="425a" id="425a" data-align="center" class="graf--h4"><em class="markup--em markup--h4-em">Better Student Journalism</em></h4>
```
Comment (gijsk) [inline test/test-pages/medium-1/expected.html]: Funny, here we should really be taking out the `h4` *and* the `h2`. Anyway, we're not making things worse, so...

File: test/test-pages/tmz-1/expected.html [hunk 2]
```diff
@@ -1,6 +1,7 @@
 <div id="readability-page-1" class="page">
     <div class="post single-post" id="post-2015_02_26_lupita-nyongo-pearl-dress-stolen-oscars">
         <p class="headline">
+            <h2 class="hf1">Lupita Nyong'o</h2>
```
Comment (gijsk) [inline test/test-pages/tmz-1/expected.html]: Maybe file a followup to remove things with class/id containing `headline`?

In fact, this `h2` is clearly a substring of the title in the metadata, so why isn't it being removed?

File: Readability.js [hunk 3]
```diff
@@ -482,10 +483,18 @@ Readability.prototype = {
       this._cleanMatchedNodes(topCandidate, /share/);
     });
 
-    // If there is only one h2, they are probably using it as a header
-    // and not a subheader, so remove it since we already have a header.
-    if (articleContent.getElementsByTagName('h2').length === 1)
-      this._clean(articleContent, "h2");
+    // If there is only one h2 and its text content substantially equals article title,
+    // they are probably using it as a header and not a subheader,
+    // so remove it since we already have a header.
```
Comment (gijsk) [inline Readability.js]: Nit: ... since we already extract the title separately.

**Comment by evanxd** [comment]
Thanks for reviewing. :)

**Comment by gijsk** [comment]
Can you file the followup to remove the `.headline` things, @evanxd ? :-)

**Comment by evanxd** [comment]
@gijsk , sure.
Sorry for replying lately. I was in vacation at that time.

**Comment by evanxd** [comment]
The followup issue: https://github.com/mozilla/readability/issues/349

**Pull request merged**